### PR TITLE
php7 typehints for Elastica\Transport namespace

### PR DESCRIPTION
**Pull request by osavchenko**
See #1548

### CODE REVIEW

File: lib/Elastica/Transport/AwsAuthV4.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 class AwsAuthV4 extends Guzzle
 {
-    protected function _getGuzzleClient($baseUrl, $persistent = true, Request $request)
+    protected function _getGuzzleClient($baseUrl, $persistent = true, Request $request): Client
```
Comment (massimilianobraglia) [inline lib/Elastica/Transport/AwsAuthV4.php]: missing `string` and `bool`

File: lib/Elastica/Transport/NullTransport.php [hunk 2]
```diff
@@ -26,9 +27,7 @@ class NullTransport extends AbstractTransport
     /**
      * Set response object the transport returns.
      *
-     * @param \Elastica\Response $response
-     *
-     * @return $this
+     * @return Response
```
Comment (massimilianobraglia) [inline lib/Elastica/Transport/NullTransport.php]: Missing `: Response` return type

File: lib/Elastica/Transport/AwsAuthV4.php (hunk 1 above)
Comment (osavchenko) [reply lib/Elastica/Transport/AwsAuthV4.php]: With that typehints it's incompatible with parent class

File: lib/Elastica/Transport/NullTransport.php (hunk 2 above)
Comment (osavchenko) [reply lib/Elastica/Transport/NullTransport.php]: By default, it trying to return null, but nullable typehints allowed only in php 7.1

Comment (massimilianobraglia) [reply lib/Elastica/Transport/NullTransport.php]: Can't you use the method `generateDefaultResponse` in order to get a Response object?

File: lib/Elastica/Transport/AwsAuthV4.php (hunk 1 above)
Comment (massimilianobraglia) [reply lib/Elastica/Transport/AwsAuthV4.php]: You should typehint the parent class, no? That class belongs to `Elastica\Transport` namespace

Comment (osavchenko) [reply lib/Elastica/Transport/AwsAuthV4.php]: Ah yes, missed that. Thank you!

File: lib/Elastica/Transport/NullTransport.php (hunk 2 above)
Comment (osavchenko) [reply lib/Elastica/Transport/NullTransport.php]: Sure

**Comment by p365labs** [comment]
you just need to execute the linter : 
```make lint```

from our command line.

**Comment by p365labs** [comment]
tnx @osavchenko !

**Pull request merged**